### PR TITLE
[dom-gpu,daint-gpu] Refactoring of TensorFlow 1.2.1 

### PR DESCRIPTION
**Pull request by mschoengens**
with all its dependencies to work with Cray 17.08 PE and the new cray-python module.

I do not know whether the change in  `easybuild/cray_external_modules_metadata.cfg` is required or has any unintentional effects on other modules. Please let me know.

### CODE REVIEW

File: easybuild/cray_external_modules_metadata.cfg [hunk 1]
```diff
@@ -142,7 +142,7 @@ version = 3.7.6.0
 prefix = PETSC_DIR
 
 [cray-python/17.06.1]
-name = CrayPython
+name = Python
```
Comment (victorusu) [inline easybuild/cray_external_modules_metadata.cfg]: Can you merge with master? 
This is already in the master branch.

File: easybuild/easyconfigs/p/protobuf/protobuf-3.1.0-CrayGNU-17.08-python3.eb [hunk 2]
```diff
@@ -0,0 +1,37 @@
+easyblock = 'PythonPackage'
+
+name = 'protobuf'
+version = '3.1.0'
+
+py_maj_ver = '3'
```
Comment (victorusu) [inline easybuild/easyconfigs/p/protobuf/protobuf-3.1.0-CrayGNU-17.08-python3.eb]: What do you think about using only the `req_py_majver` and `req_py_minver`?

File: jenkins-builds/6.0.UP04-17.08-gpu [hunk 3]
```diff
@@ -50,6 +50,7 @@
  Score-P-3.1-CrayIntel-17.08.eb
  Score-P-3.1-CrayPGI-17.08.eb
  Spark-1.6.0.eb                                     --set-default-module
+ TensorFlow-1.2.1-CrayGNU-17.08-cuda-8.0.54-python3.eb
```
Comment (victorusu) [inline jenkins-builds/6.0.UP04-17.08-gpu]: Can you make this version default?

File: easybuild/easyconfigs/p/protobuf/protobuf-3.1.0-CrayGNU-17.08-python3.eb (hunk 2 above)
Comment (mschoengens) [reply easybuild/easyconfigs/p/protobuf/protobuf-3.1.0-CrayGNU-17.08-python3.eb]: Actually I have thought about this as well, but then I have convert to str everytime I use it, and that is quite often. Definitely decreases code readability.

On the other hand, req_py_* is only used internally.

Comment (victorusu) [reply easybuild/easyconfigs/p/protobuf/protobuf-3.1.0-CrayGNU-17.08-python3.eb]: 👍

**Comment by victorusu** [comment]
test this please

**Comment by victorusu** [comment]
test this please

**Comment by lucamar** [comment]
ok to test

**Comment by gppezzi** [comment]
Thanks @mschoengens and @victorusu !

**Pull request merged**